test/test-lib/fake-device.ts

```diff
@@ -70,11 +70,17 @@ export async function provisionDevice(
 			belongs_to__application: appId,
 			uuid: uuid.v4().replace(/\-/g, '').toLowerCase(),
 			device_type: deviceType,
-			os_version: '2.38.0+rev1',
-			supervisor_version: 'v10.0.0',
 		})
 		.expect(201);
 
+	await supertest(app, admin)
+		.patch(`/resin/device(${device.id})`)
+		.send({
+			os_version: '2.38.0+rev1',
+			supervisor_version: '5.0.1',
+		})
+		.expect(200);
+
```

Comment:
I thought the aim of the PR was to allow the supervisor release to be set when the device `PATCH`es the state endpoint, and subsequently patches the device resource? This step here doesn't seem to make sense since this action isn't performed by the supervisor directly... what am I missing?

Reply:
We discussed this internally. tl;dr is that only the new test case will PATCH in this way (but the code will be removed from the initial POST). additionally we will explicitly test a PATCH to the state endpoint.

Reply:
This is now implemented.